CMSgov/mint-app
# [EASI-2454] - Document upload restriction

# EASI-000

## Changes and Description

- Added `restricted` boolean to BE document table and schema
- Added resolver to check call access control method `IsCollaboratorModelPlanID`
- SQL statement to only return non restricted documents and implemented as a part of the resolver if not a collaborator
- Updated BE tests and seeded data
- FE Document upload form change to capture restricted
- Document table column added


![Screenshot 2022-11-09 at 2 50 44 PM](https://user-images.githubusercontent.com/95709965/200927395-81625ce0-79ca-4a97-95a4-9a48304fe2a1.png)

![Screenshot 2022-11-09 at 2 51 19 PM](https://user-images.githubusercontent.com/95709965/200927497-37f540de-1a07-4f07-ab48-371672ede6da.png)

![Screenshot 2022-11-09 at 2 51 45 PM](https://user-images.githubusercontent.com/95709965/200927562-2c38061a-3b79-4dbe-9c63-41c3dd3d6a32.png)

## Notes

- I haven't touched this other resolver `PlanDocumentRead` as it is not utilized anywhere in the FE.  The query is written, I believe deprecated?


## How to test this change

- Create documents with both restricted/cost information true and false
- Verify the documents appear with the `Visibility` column populated correctly
- Log in as not a collaborator and not an assessment user
- Navigate to the same model plan in the document -> read only section
- Verify the restricted document does not appear in the document table


## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
